docs/NNNN-schema-registry.md

```diff
@@ -0,0 +1,195 @@
+
+The values of digestAlgorithm and identifierDigestAlgorithm should not be changed once the registry is initialised. If changing the digest is unavoidable, all existing entries in the registry must be updated to the new algorithm(s).
+
+Two possible implementation cases are considered:
+
+### Where an OCFL root is being generated as a snapshot for export or archival purposes.
+
+In this case, it is reasonable to build or check the completeness of the schema registry as each OCFL object is processed for backup. Then the exported root will contain all necessary schemata.
+
+### Where an OCFL root provides the persistence layer for a live repository 
+
+In this case, a background process may discover or be informed of any new OCFL objects/versions. This process can asynchronously process the object contents, retrieve the required schemata and perform necessary registrations with the schema registry.
+
+The values of digestAlgorithm and identifierDigestAlgorithm should not be changed once the registry is initialised. If changing the digest is unavoidable, all existing entries in the registry must be updated to the new algorithm(s).
+
+### Registering a schema with the extension
+
+* New OCFL objects/versions must be inspected for JSON, XML files and references to external schema ($schema / XML-DTD etc) are extracted. This may occur periodically or be triggered by creation of a new or updated OCFL object.
+* For each referenced schema:
+  * The local filename is derived by hashing the schema's identifier using the configured schemaDigestAlgorithm.
+  * The manifest object in schemata_inventory.json must be checked for this key. 
+    * If this identifier is not already present in the schema repository:
+      * A copy of the remote schema must be retrieved, and stored in the schemata/ directory with the derived local filename.
+      * The local copy's digest must be calculated with the configured digestAlgorithm.
+      * The schemata_inventory.json must updated:
+        * a new key must be created in the manifest object, with the properties 'digest' and 'identifier'.
+      * The schema_inventory.json’s sidecar file must also be updated with the appropriate checksum.
+    * If the schema's checksum is already registered, the identifiers must be compared to exclude the possibility of hash-collision (malicious or otherwise).
+      * If the identifiers match, the schema is already registered and no further action needs to be taken.
+      * If the identifiers differ, an exception must be raised.
```

Comment:
To me, this is the important bit. Personally, I'd prioritize this description over the implementation suggestions.